cli/cmd/ps/ps_services.go

```diff
@@ -114,20 +115,26 @@ func (p *Ps) services(ctx *clicontext.CLIContext) error {
 		}
 	}
 
-	for _, service := range services {
+	for _, service := range svcs {
 		if service.(*riov1.Service).Spec.Template && !p.A_All {
 			continue
 		}
 
 		weight := 0
 		if service.(*riov1.Service).Status.ComputedWeight != nil && *service.(*riov1.Service).Status.ComputedWeight > 0 {
-			totalWeight := 0
-			for _, subService := range services {
-				if service.(*riov1.Service).Spec.App == subService.(*riov1.Service).Spec.App && subService.(*riov1.Service).Status.ComputedWeight != nil {
-					totalWeight += *subService.(*riov1.Service).Status.ComputedWeight
+			app, _ := services.AppAndVersion(service.(*riov1.Service))
```

Comment:
Hmm, app will not be empty in this case.

Reply:
Ohh, I bet its just falling down to next check and that check works correctly now. I'll remove.